lib/cli/interactive-setup/index.js

```diff
@@ -74,9 +74,9 @@ module.exports = async (context) => {
     const { provider } = context.serverless.service;
     if (!devModeRuntimeCompatibility.includes(provider.runtime)) {
       log.error('This services runtime is not currently supported by Serverless Console Dev Mode.');
-    } else {
-      context.isConsoleDevMode = true;
+      context.shouldSkipDevMode = true;
```

Comment:
Would it be more clear to use a skip property though? If we just flip the `isConsole` flag we risk steps that are not intended to be triggered getting triggered and we will need to step through the different states to figure out when isConsole should be true or false.

Reply:
I also changed this so that we would consistently use `shouldSkipDevMode` when we fail due to some calculated issue. 

I am kind of looking at it like `isConsole` tells us if `--console` was set, `isConsoleDevMode` tells us if `--dev` was set, and `shouldSkipDevMode` tells us if we should just exit early.

I was thinking if we used flags like these we wouldn't have variables that mean more than one thing and we might not have to step through the logic as much to figure out what is going on.

Reply:
> What steps exactly?

The dashboard login one appears to be activated.

> My understanding is that it's not some unprecedented special case. but that simply if there are no functions for which dev mode can be applied, we should not apply it. And the applicability of steps is marked with booleans as isConsole and isConsoleDevMode

I don't think it is unprecedented I just know I am going to have a hard time keeping track of the logic if we change the meaning of the different flags as we use them so I wanted to make on flag that could be used in the case we want to skip a step.

For example, if I simply uncheck the `isConsole` flag here the dashboard login flag is run for some reason 👇 
<img width="936" alt="Screenshot 2023-03-16 at 3 30 39 PM" src="https://user-images.githubusercontent.com/1753824/225745800-dd30659e-6704-460a-b3c5-858065641764.png">

Reply:
> The dashboard login one appears to be activated.

I'll rather fix that by introducing `isDashboard` to context instead (setting it with `!options.console && !options.dev`) and ten rely on this instead of `!isConsole` in dashboard steps.

> so I wanted to make on flag that could be used in the case we want to skip a step.

Skip flag, is another enable/disable flag. While we already have enable/disable flag for dev mode. It's `isConsoleDevMode` (logically it could be also named `shouldEnableDevMode`). Having two boolean flags for same thing doesn't feel right.

Reply:
I am at it like the isConsole tells us if `--console` was set, isConsoleDevMode tells us if `--dev` was set, and shouldSkipDevMode tells us if we should just exit early.

If we use isConsoleDevMode and isConsole as skip flag in one scenario then a context flag in another we will have Booleans that mean two things. The main issue I see with this is that the only way to find out both of these meanings is to read these specific blocks of code. 

I am happy to leave it as your call I just want to make sure it is clear why I think reusing the existing flags is confusing to me.

Reply:
> I am at it like the isConsole tells us if --console was set, isConsoleDevMode tells us if --dev

It's `options.console` that tells us if `--console` was set, and `options.dev` tells if `--dev` is set.
Both `isConsole` and `isConsoleDevMode` are result of the complete preliminary check, that tell wether we should bother with _console_ related and _console dev mode_ related steps.

If `isConsole` would be just as an alias for `options.console` it wouldn't be introduced at all (as why not simply check `options.console`?)

> If we use isConsoleDevMode and isConsole as skip flag in one scenario then a context flag in another we will have Booleans that mean two things.

Where exactly do you see that their meaning will differ?

Reply:
If that is not how those flags are meant to be viewed then I will just go ahead and refactor to use only `isConsole` and `isConsoleDevMode`.

Reply:
Yes I think it's best to stick the convention we're using right now, that if we need a more sophisticated applicability check for a group of steps, introduce `is<stepGroupName>` variable.

We have one for _console_ and _consoleDevMode_ steps. This case shown that we probably also need to introduce one for _dashboard_.